kubectl-minio/cmd/resources/tenant.go

```diff
@@ -134,5 +141,56 @@ func NewTenant(opts *TenantOptions, user *v1.Secret) (*miniov2.Tenant, error) {
 			},
 		},
 	}
+	if opts.EnableAuditLogs {
+		t.Spec.Log = getAuditLogConfig(opts)
+	}
 	return t, t.Validate()
 }
+
+func getAuditLogConfig(opts *TenantOptions) *miniov2.LogConfig {
+	diskSpace := int64(opts.AuditLogsDiskSpace) * humanize.GiByte
+	var logSearchStorageClass *string
+	if opts.AuditLogsStorageClass != "" {
+		logSearchStorageClass = &opts.AuditLogsStorageClass
+	}
+	// the audit max cap cannot be larger than disk size on the DB, else it won't trim the data
+	auditMaxCap := 10
+	if (diskSpace / humanize.GiByte) < int64(auditMaxCap) {
+		auditMaxCap = int(diskSpace / humanize.GiByte)
+	}
+	logConfig := createLogConfig(diskSpace, auditMaxCap, opts.Name, logSearchStorageClass)
+	if opts.AuditLogsImage != "" {
+		logConfig.Image = opts.AuditLogsImage
+	}
+	if opts.AuditLogsPGImage != "" {
+		logConfig.Db.Image = opts.AuditLogsPGImage
+	}
+	if opts.AuditLogsPGInitImage != "" {
+		logConfig.Db.InitImage = opts.AuditLogsPGInitImage
+	}
+	return logConfig
+}
+
+func createLogConfig(diskSpace int64, auditMaxCap int, name string, storage *string) *miniov2.LogConfig {
+	return &miniov2.LogConfig{
+		Audit: &miniov2.AuditConfig{DiskCapacityGB: &auditMaxCap},
+		Db: &miniov2.LogDbConfig{
+			VolumeClaimTemplate: &v1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name + "-log",
+				},
+				Spec: v1.PersistentVolumeClaimSpec{
+					AccessModes: []v1.PersistentVolumeAccessMode{
+						v1.ReadWriteOnce,
+					},
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							v1.ResourceStorage: *resource.NewQuantity(diskSpace, resource.DecimalExponent),
```

Comment:
Let's confirm this line first.

when deploying audit log via yamls operator will assign the storage using `resource.BinarySI`.
Ideally we should abstract the logic of the `NewForLogDb` function in https://github.com/minio/operator/blob/master/pkg/resources/statefulsets/log.go#L147 and use the same logic for both deployment types yamls and kubectl plugin